ballerina/types.bal

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Provides a set of configurations to control the behaviours when communicating with a schema registry.
+#
+# + hostName - The host name of the Active Directory server
+# + port -  The port of the Active Directory server
+# + domainName -  The domain name of the Active Directory
+# + password - The password of the Active Directory
+public type ConnectionConfig record {|
+    string hostName;
+    int port;
+    string domainName;
+    string password;
+|};
+
+# LDAP response type.
+#
+# + matchedDN - The matched DN from the response
+# + resultStatus - The operation status of the response
+# + operationType - The protocol operation type
+public type LDAPResponse record {|
+    string matchedDN;
+    Status resultStatus;
+    string operationType;
+|};
+
+
+# A record for an entry that represents a person.
+#
+# + objectClass - object class of the person
+# + sn - surname of the person
+# + cn - common name of the person
+# + userPassword - password of the person
+# + telephoneNumber - telephone number of the person
+public type Person record {
+    string|string[]|ObjectClass|ObjectClass[] objectClass;
+    string sn;
+    string cn;
+    string userPassword?;
+    string telephoneNumber?;
+};
+
+enum ObjectClass {
```

Comment:
Shouldn't this be a public enum that is visible to users?

Reply:
Fixed in [cdde708](https://github.com/ballerina-platform/module-ballerina-ldap/pull/45/commits/cdde70852f32464ef384105c5366885fa9486d98)